Core/src/AnonymousCredentials.php

```diff
@@ -83,7 +90,7 @@ public function updateMetadata(
         $authUri = null,
         callable $httpHandler = null
     ) {
-        return $metadata;
+        return $metadata + $this->authHeader;
```

Comment:
Why are we doing this? This is a change in behavior that shouldn't be required. Especially since there can potentially be other custom credentials types that _don't_ do this (because they're defined by a customer)